### PR TITLE
Add a tool to generate snippet markdown for docfx to consume

### PR DESCRIPTION
**Pull request by jskeet**
(Better suggestions for the namespace are welcome. This isn't going to be a public API, so doesn't need
to be stable.)

Additionally, tweak the existing snippets and add a new one, partly as part of testing the tool.

### CODE REVIEW

File: tools/Google.GCloud.Tools.GenerateSnippetMarkdown/Program.cs [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+    {
+        private static readonly Regex StartSnippetTag = new Regex(@"^\s*// \<(?<name>[\w\.]+)>\s*$", RegexOptions.Compiled);
+
+        private static void Main(string[] args)
+        {
+            string root = DetermineRootDirectory();
+            if (root == null)
+            {
+                Console.WriteLine("Unable to determine root directory. Please run within gcloud-dotnet repository.");
+                return;
```
Comment (ivannaranjo) [inline tools/Google.GCloud.Tools.GenerateSnippetMarkdown/Program.cs]: Can we use exit codes here? So you can signal error/success for scripting this tool.

File: tools/Google.GCloud.Tools.GenerateSnippetMarkdown/Program.cs [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+            using (var outputMarkdown = File.CreateText(outputFile))
+            {
+                // Search every line, in every source file, in every project.
+                // If we find a snippet, e.g. // <snippetName>, then find all the UIDs from YAML
+                // files that are for that method name, scoped by the project/source names.
+                var query = from project in Directory.GetDirectories(snippetsDir)
+                            // Path.GetFileName just takes the last part of the name; it doesn't know that it's a directory.
+                            let projectName = TrimSuffix(Path.GetFileName(project), ".Snippets")
+                            from sourceFile in Directory.GetFiles(project, "*.cs")
+                            let sourceRelativeToOutput = $"../../snippets/{Path.GetFileName(project)}/{Path.GetFileName(sourceFile)}"
```
Comment (ivannaranjo) [inline tools/Google.GCloud.Tools.GenerateSnippetMarkdown/Program.cs]: Nit: you were very careful about using `Path.Combine()` above, perhaps we could do the same here?

File: tools/Google.GCloud.Tools.GenerateSnippetMarkdown/Program.cs (hunk 1 above)
Comment (jskeet) [inline tools/Google.GCloud.Tools.GenerateSnippetMarkdown/Program.cs]: Sure, that's easy enough to do.

File: tools/Google.GCloud.Tools.GenerateSnippetMarkdown/Program.cs (hunk 2 above)
Comment (jskeet) [inline tools/Google.GCloud.Tools.GenerateSnippetMarkdown/Program.cs]: Well, not with Combine. We basically need a way of getting a relative path - definitely using forward slashes - from outputFile to snippetsDir. Will have a think.

**Comment by jskeet** [comment]
(Now added the exit codes as requested by Ivan.)

**Comment by chrisdunelm** [comment]
LGTM

**Pull request merged**